src/main/java/io/github/mzmine/datamodel/impl/MsdkScanWrapper.java

```
@@ -104,12 +106,12 @@ public double[] getIntensityValues(@NotNull double[] dst) {
 
   @Override
   public double getMzValue(int index) {
-    return mzs[index];
+    return mzs.array()[index];
```

This method should not be called here. Wrapper should just throw an exception. 
Also .array() is unsafe. 
Should use a get method on doublebuffer and use index or even better load the spectra into an array if used in a loop.
```
    throw new UnsupportedOperationException(
        "Unsupported operation. MSDK scan uses float array and the conversion in this method is not efficient.");
```